lib/debug.ml

```diff
@@ -122,26 +122,28 @@ let format include_time brand priority message =
 let print_debug = ref false
 let log_to_stdout () = print_debug := true
 
+module BrandLevelPair = struct
+  type t = string * Syslog.level
+  let compare = Pervasives.compare
+end
+module BrandLevelPairSet = Set.Make(BrandLevelPair)
+
 let loglevel_m = Mutex.create ()
-let logging_disabled_for : (string * Syslog.level, unit) Hashtbl.t = Hashtbl.create 0
+let logging_disabled_for = ref BrandLevelPairSet.empty
 let default_loglevel = Syslog.Debug
 let loglevel = ref default_loglevel
 
 let disabled_modules () =
-  Mutex.execute loglevel_m (fun () ->
-      Hashtbl.fold (fun key _ acc -> key :: acc) logging_disabled_for []
-    )
+    BrandLevelPairSet.fold (fun brandlevel acc -> brandlevel :: acc) !logging_disabled_for []
 
 let is_disabled brand level =
-  Mutex.execute loglevel_m (fun () ->
-      Syslog.is_masked ~threshold:!loglevel level ||
-      Hashtbl.mem logging_disabled_for (brand, level)
-    )
+    Syslog.is_masked ~threshold:!loglevel level ||
+    BrandLevelPairSet.mem (brand, level) !logging_disabled_for
```

Comment:
Looks good, previously we used a mutex because we were iterating through a mutable data structure that could be changed by another thread.
Now that the data structure is immutable we do not need locks anymore: passing it to the function "takes a snapshot" effectively, so even if it changes it is safe to iterate on it.

We still need locks when modifying it, so that operations that add and remove elements do not cause race conditions.